securesign/secure-sign-operator
# feat: add secrets, add cert creation, make optional

Pr to allow the operator to either create and use certs, or use certs created by user.

## Testing without cert and secrets
1. Launch and authenticate with cluster
2. Install Keycloak on the cluster
   1. `oc create -k https://github.com/securesign/sigstore-ocp/keycloak/operator/base`
   2. `oc create -k https://github.com/securesign/sigstore-ocp/keycloak/resources/base`
3. Run the operator
   1. `make install && make run` 
4. Add keycloak url to `config/samples/rhtas_v1alpha1_securesign.yaml`
5. Create a namespace: `oc new-project securesign`
6. Apply the yaml file: `oc apply -f config/samples/rhtas_v1alpha1_securesign_cert_create.yaml -n securesign`
7. Sign and verify

## Testing with cert and secrets
1. As Above (1-5)
2. Create secrets in namespace
 ``` 
oc -n securesign create secret generic fulcio-secret-rh --from-file=private=file_ca_key.pem --from-file=public=file_ca_pub.pem --from-file=cert=fulcio-root.pem --from-literal=password=secure  && oc -n securesign create secret generic rekor-private-key --from-file=private=rekor_key.pem
  ```
3. Apply the yaml file: `oc apply -f config/samples/rhtas_v1alpha1_securesign.yaml -n securesign`
4. Sign and verify


Completes https://github.com/securesign/secure-sign-operator/issues/1
